src/hooks/useConnectToValidChain.tsx

```diff
@@ -0,0 +1,31 @@
+import { useNetwork } from "wagmi";
+
+/**
+ * @remarks if user is not connected to supported chains (Mainnet 1, or Goerli 5), this hook will prompt user to switch to Mainnet
+ * @returns void
+ */
+export const useConnectToValidChain = async () => {
+  const { chain } = useNetwork();
+
+  if (chain?.id === 1 || chain?.id === 5) {
+    return;
+  }
+
+  try {
+    await window.ethereum.request({
```

Comment:
may not work for all wallets - better to use the hook mentioned at the start of this review.